cedar-policy-validator/src/type_error.rs

```diff
@@ -212,7 +212,7 @@ impl std::error::Error for TypeError {}
 pub enum TypeErrorKind {
     /// The typechecker expected to see a subtype of one of the types in
     /// `expected`, but saw `actual`.
-    #[error("Unexpected type. Expected {} but saw {}",
+    #[error("unexpected type: expected {} but saw {}",
```

Comment:
should types be surrounded with backticks?

Reply:
I decided not to surround types with backticks since they are not verbatim user input, and I felt that `expected bool but saw string` was just as clear as ``expected `bool` but saw `string` ``.  Open to second opinions.

Note: Your comment made me realize that we weren't following the rules when printing types (we were printing `entity of type Foo` rather than ``entity of type `Foo` ``). Fixing this required changing a bunch of the corpus tests, which is why the diff in my recent commit is so large.

Reply:
For reference, here is how we currently print types:

* `bool`
* `long`
* `string`
* `set`
* `record`
* `(entity of unspecified type)`
* ``(entity of type `Foo`)`` where `Foo` is some user-specified name